lib/puppet/pops/model/ast.rb

```diff
@@ -4,19 +4,19 @@ module Puppet
 module Pops
 module Model
 class PopsObject
-  def self._plocation
-    @_plocation ||= (loc = Puppet::Util.path_to_uri("#{__FILE__}"); URI("#{loc}?line=#{__LINE__.to_i - 3}"))
+  def self._pcore_location
```

Comment:
unless we turn this into something generic, but then that generic thing is not "location" but some mechanism that provides a link back to document information in the root of the model instance